src/main/java/wis/Ui.java

```diff
@@ -0,0 +1,167 @@
+package wis;
+
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.InputMismatchException;
+
+import wis.task.Task;
+import wis.task.Todo;
+import wis.task.Deadline;
+import wis.task.Event;
+import wis.task.TaskList;
+import wis.util.InputParser;
+import wis.util.WisException;
+import wis.util.Pair;
+import wis.util.Printer;
+
+/**
+ * Part of the chat box which deals with user interation.
+ */
+public class Ui {
+    private String input;
+
+    public Ui(String input) {
+        this.input = input;
+    }
+
+    public void setInput(String input) {
+        this.input = input;
+    }
+
+    /**
+     * Parses user input for action, and performs the action.
+     *
+     * @param tasks Task list storing all tasks.
+     * @return True if and only if user input signals exit.
+     */
+    public boolean parseInput(TaskList tasks) {
+        String[] words = input.split(" ");
+        Action action = InputParser.parseAction(input, words);
+        switch (action) {
+        case NONE:
+            break;
+        case ADD_TODO:
+            addTodo(tasks);
+            break;
+        case ADD_DEADLINE:
+            addDeadline(tasks);
+            break;
+        case ADD_EVENT:
+            addEvent(tasks);
+            break;
+        case MARK:
+            mark(words, tasks);
+            break;
+        case UNMARK:
+            unmark(words, tasks);
+            break;
+        case DELETE:
+            delete(words, tasks);
+            break;
+        case LIST:
+            Printer.printActionAttach(Action.LIST, tasks);
+            break;
+        case FIND:
+            find(words, tasks);
+            break;
+        case BYE:
+            return true;
+        case INVALID:
+            WisException.actionExceptionHandler(Action.INVALID);
+            break;
+        default:
+            throw new IllegalArgumentException("Illegal action argument provided.\n");
+        }
```

Comment:
Perhaps a bit of nitpicking to clear the indentation on switch blocks as per the conventions 😄